File: backend/features/notifications/notificationRoutes.js
// backend\features\notifications\notificationRoutes.js
const express = require("express");
const router = express.Router();
const pool = require("../../config/database");
const { authenticate } = require("../../shared/middleware/tokenMiddleware");


// PATCH /notifications/read-all — mark all read
router.patch("/read-all", authenticate, async (req, res) => {
  try {
    await pool.query(
      `UPDATE notifications SET is_read = TRUE WHERE recipient_user_id = $1`,
      [req.user.user_id]
    );
    res.json({ success: true });
  } catch (err) {
    res.status(500).json({ success: false, message: err.message });
  }
});
// GET /notifications — fetch my notifications
router.get("/", authenticate, async (req, res) => {
  try {
    const result = await pool.query(
      `SELECT 
         n.id, n.sender_name, n.type, n.title, n.message, 
         n.link_to, n.is_read, n.created_at,
         u.profile_picture AS sender_avatar
       FROM notifications n
       LEFT JOIN users u ON n.sender_user_id = u.user_id
       WHERE n.recipient_user_id = $1
       ORDER BY n.created_at DESC
       LIMIT 50`,
      [req.user.user_id]
    );
    const unread = result.rows.filter((n) => !n.is_read).length;
    res.json({ success: true, data: result.rows, unread });
  } catch (err) {
    res.status(500).json({ success: false, message: err.message });
  }
});

// PATCH /notifications/:id/read — mark one read
router.patch("/:id/read", authenticate, async (req, res) => {
  try {
    await pool.query(
      `UPDATE notifications SET is_read = TRUE
       WHERE id = $1 AND recipient_user_id = $2`,
      [req.params.id, req.user.user_id]
    );
    res.json({ success: true });
  } catch (err) {
    res.status(500).json({ success: false, message: err.message });
  }
});

router.post("/push-token", authenticate, async (req, res) => {
  try {
    const { push_token } = req.body;
    await pool.query(
      `UPDATE users SET push_token = $1 WHERE user_id = $2`,
      [push_token, req.user.user_id]
    );
    res.json({ success: true });
  } catch (err) {
    res.status(500).json({ success: false, message: err.message });
  }
});
module.exports = router;

File: backend/features/notifications/notification_routes.py
# # ================================================================================
# # FILE: backend/features/notifications/notification_routes.py
# # ================================================================================

# from fastapi import APIRouter, HTTPException, Request

# import config.database as db
# from shared.middleware.token_middleware import authenticate

# router = APIRouter()


# def get_pool():
#     if db.pool is None:
#         raise RuntimeError("Database pool is not initialized")
#     return db.pool


# @router.patch("/read-all")
# async def read_all(request: Request):
#     try:
#         user_id = request.state.user["user_id"]
#         await get_pool().execute(
#             "UPDATE notifications SET is_read = TRUE WHERE recipient_user_id = $1",
#             user_id,
#         )
#         return {"success": True}
#     except Exception as err:
#         raise HTTPException(status_code=500, detail={"success": False, "message": str(err)})


# @router.get("/")
# async def get_notifications(request: Request):
#     try:
#         user_id = request.state.user["user_id"]
#         rows = await get_pool().fetch(
#             """
#             SELECT
#               n.id, n.sender_name, n.type, n.title, n.message,
#               n.link_to, n.is_read, n.created_at,
#               u.profile_picture AS sender_avatar
#             FROM notifications n
#             LEFT JOIN users u ON n.sender_user_id = u.user_id
#             WHERE n.recipient_user_id = $1
#             ORDER BY n.created_at DESC
#             LIMIT 50
#             """,
#             user_id,
#         )
#         data   = [dict(r) for r in rows]
#         unread = sum(1 for n in data if not n["is_read"])
#         return {"success": True, "data": data, "unread": unread}
#     except Exception as err:
#         raise HTTPException(status_code=500, detail={"success": False, "message": str(err)})


# @router.patch("/{notification_id}/read")
# async def mark_read(notification_id: int, request: Request):
#     try:
#         user_id = request.state.user["user_id"]
#         await get_pool().execute(
#             "UPDATE notifications SET is_read = TRUE WHERE id = $1 AND recipient_user_id = $2",
#             notification_id, user_id,
#         )
#         return {"success": True}
#     except Exception as err:
#         raise HTTPException(status_code=500, detail={"success": False, "message": str(err)})


# @router.post("/push-token")
# async def save_push_token(request: Request):
#     try:
#         user_id = request.state.user["user_id"]
#         body    = await request.json()
#         await get_pool().execute(
#             "UPDATE users SET push_token = $1 WHERE user_id = $2",
#             body.get("push_token"), user_id,
#         )
#         return {"success": True}
#     except Exception as err:
#         raise HTTPException(status_code=500, detail={"success": False, "message": str(err)})